data/daos/implementations/ArrayListStudentDao.java

```diff
@@ -0,0 +1,37 @@
+package data.daos.implementations;
+
+import data.daos.StudentDao;
+import data.daos.exceptions.NoSuchStudentExist;
+import logic.models.Student;
+
+import java.util.ArrayList;
+
+public class ArrayListStudentDao implements StudentDao {
+    private final ArrayList<Student> students;
+
+    public ArrayListStudentDao() {
+        students = new ArrayList<>();
+    }
+
+    @Override
+    public void add(Student student) {
+        students.add(student);
+    }
+
+    @Override
+    public void remove(String idToRemove) {
+        students.remove(students.get(students.indexOf(get(idToRemove))));
```

Comment:
Use the Java stream API to make it more readable. First try to find the student to remove using `filter`, if a student is present, remove it. Else throw an exception, that indicates, that no student with the given id exists.

Reply:
Yeah Raimund Will use from now on. Updated Raimund.